core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -865,25 +865,41 @@ public void deleteInode(RpcContext rpcContext, LockedInodePath inodePath, long o
     }
   }
 
+  private boolean checkPinningValidity(Set<String> pinnedLocations) {
+    List<String> mediaList = ServerConfiguration.getList(
+        PropertyKey.MASTER_TIERED_STORE_GLOBAL_MEDIA, ",");
+    for (String medium : pinnedLocations) {
+      if (!mediaList.remove(medium)) {
+        // mediaList does not contains medium
+        return false;
+      }
+    }
+    return true;
+  }
   /**
    * Sets the pinned state of an inode. If the inode is a directory, the pinned state will be set
    * recursively.
    *
    * @param rpcContext the rpc context
    * @param inodePath the {@link LockedInodePath} to set the pinned state for
    * @param pinned the pinned state to set for the inode (and possible descendants)
+   * @param mediumTypes the list of pinned media that that the file can reside in
    * @param opTimeMs the operation time
    * @throws FileDoesNotExistException if inode does not exist
    */
   public void setPinned(RpcContext rpcContext, LockedInodePath inodePath, boolean pinned,
-      long opTimeMs) throws FileDoesNotExistException, InvalidPathException {
+      List<String> mediumTypes, long opTimeMs)
```

Comment:
Could `mediumTypes` be of type `Set<String>` instead?

Reply:
The user inputs a list, somewhere we have to do the list to set conversion since we are using set internally, where do you think this conversion should happen?

Reply:
The conversion from user input to set should happen immediately since we only use set semantics